src/spec/integration/testhelpers.coffee

```diff
@@ -101,3 +101,17 @@ exports.setupProductType = (client, productType, product) ->
       .then (result) -> Promise.resolve result.body # returns product
     else
       Promise.resolve pt # returns productType
+
+
+exports.ensureCategories = (client, categoryList) ->
+  console.log 'About to cleanup categories...'
+  client.categories
+  .perPage(30)
+  .process (res) ->
+    Promise.map res.body.results, (category) ->
+      client.categories.byId(category.id).delete(category.version)
+  .then (result) ->
+    console.log "Deleted #{_.size result} categories, creating new one"
+    Promise.map categoryList, (category) ->
+      client.categories.create(category)
+      .then (result) -> Promise.resolve(result.body)
```

Comment:
you can just return `result.body`. The return value of `.then()` is always a promise that resolves to the return value of the inner callback function.